pkg/vm/engine/tae/txn/txnbase/txnmgr.go

```diff
@@ -129,39 +135,65 @@ func (mgr *TxnManager) GetTxn(id uint64) txnif.AsyncTxn {
 	return mgr.IDMap[id]
 }
 
+func (mgr *TxnManager) EnqueueFlushing(op any) (err error) {
+	_, err = mgr.PreparingSM.EnqueueCheckpoint(op)
+	return
+}
+
+func (mgr *TxnManager) EnqueueCommitting(op any) (err error) {
+	_, err = mgr.CommittingSM.EnqueueRecevied(op)
+	return
+}
+
+func (mgr *TxnManager) EnqueueCFlushing(op any) (err error) {
+	_, err = mgr.CommittingSM.EnqueueCheckpoint(op)
+	return
+}
+
 func (mgr *TxnManager) OnOpTxn(op *OpTxn) (err error) {
-	_, err = mgr.EnqueueRecevied(op)
+	_, err = mgr.PreparingSM.EnqueueRecevied(op)
 	return
 }
 
-func (mgr *TxnManager) onPreCommit(txn txnif.AsyncTxn) {
+func (mgr *TxnManager) onPreCommitOr2PCPrepare(txn txnif.AsyncTxn) {
 	now := time.Now()
-	txn.SetError(txn.PreCommit())
+	txn.SetError(txn.PreCommitOr2PCPrepare())
 	logutil.Debug("[PreCommit]", TxnField(txn), common.DurationField(time.Since(now)))
 }
 
 func (mgr *TxnManager) onPreparCommit(txn txnif.AsyncTxn) {
 	txn.SetError(txn.PrepareCommit())
 }
 
+func (mgr *TxnManager) onPrepar2PCPrepare(txn txnif.AsyncTxn) {
+	txn.SetError(txn.Prepare2PCPrepare())
+}
+
 func (mgr *TxnManager) onPreApplyCommit(txn txnif.AsyncTxn) {
 	if err := txn.PreApplyCommit(); err != nil {
 		txn.SetError(err)
 		mgr.OnException(err)
 	}
 }
 
+func (mgr *TxnManager) onPreApply2PCPrepare(txn txnif.AsyncTxn) {
+	if err := txn.PreApplyCommit(); err != nil {
+		txn.SetError(err)
+		mgr.OnException(err)
+	}
+}
+
 func (mgr *TxnManager) onPreparRollback(txn txnif.AsyncTxn) {
 	_ = txn.PrepareRollback()
 }
 
-// TODO
 func (mgr *TxnManager) onPreparing(items ...any) {
```

Comment:
Could we separate 1PC and 2PC logic?

Reply:
1PC  and 2PC  transactions must both enter into `onPreparing `for conflict check, etc.   2PC  had been separated from 1PC  inside the `onPreparing `

Reply:
`onPreparing`  means that  do preparing work for  1PC  Commit or  2PC  Prepare